user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -134,15 +135,9 @@ public void maybeGenerateCertificates(CertManager certManager,
         } else if (clientsCaKeySecret == null) {
             throw new NoCertificateSecretException("The Clients CA Key Secret is missing");
         } else {
-            String valStr = System.getenv(ENV_VAR_CLIENTS_CA_VALIDITY);
-            String renStr = System.getenv(ENV_VAR_CLIENTS_CA_RENEWAL);
-
-            int validity = 0;
-            int renewal = 0;
-            if (valStr != null)
-                validity = Integer.parseInt(valStr);
-            if (renStr != null)
-                renewal = Integer.parseInt(renStr);
+            UserOperatorConfig config = UserOperatorConfig.fromMap(System.getenv());
```

Comment:
I'm not sure how expensive operation this is. But having it here means it can be possibly called many many times. Can't we pass it somehow from the `UserOperatorConfig` instance created at start?

@tombentley What do you think about this?

Reply:
@scholzj is right, these should be parameters to `fromCrd()` and passed from there to `maybeGenerateCertificates()`. Also `maybeGenerateCertificates()` (and `maybeGeneratePassword()`) should be `private`.

Reply:
TBH, what I meant as a solution was to just pass the today variables through the objects. Not make them static methods somewhere else. You will still do the `System.getEnv()` again and again and again although it doesn't really change. Although [SO](https://stackoverflow.com/questions/42196291/the-performance-impact-of-use-system-getenv-in-java) seems to suggest it is cheap operation. I will leave it up to @tombentley to see if that is or is not an issue.